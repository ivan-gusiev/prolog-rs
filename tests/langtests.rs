extern crate insta;
extern crate parameterized;
extern crate prolog_rs;

#[cfg(test)]
mod langtests {
    use insta::assert_display_snapshot;
    use parameterized::parameterized;
    use prolog_rs::{
        lang::parse_term,
        symbol::{SymbolTable, to_display},
        util::{case, case_dbg},
    };

    #[parameterized(input = {
        "p(Z, h(Z,W), f(W))",
        "D",
        "",
        "size(Width, Height)",
    })]
    fn test_term_parse(input: &str) {
        assert_display_snapshot!(case_dbg(
            input,
            parse_term(input, &mut (SymbolTable::new()))
        ));
    }

    #[parameterized(input = {
        "p(Z, h(Z,W), f(W))",
        "D",
        "",
        "size(Width, Height)",
    })]
    fn test_term_display(input: &str) {
        let mut symbol_table = SymbolTable::new();
        match parse_term(input, &mut symbol_table) {
            Ok(term) => assert_display_snapshot!(case(input, to_display(&term, &symbol_table))),
            Err(err) => assert_display_snapshot!(case(input, err)),
        }
    }
}
