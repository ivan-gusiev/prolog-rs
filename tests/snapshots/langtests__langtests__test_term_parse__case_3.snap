---
source: tests/langtests.rs
expression: "case_dbg(input, parse_term(input, &mut (SymbolTable::new())))"
---
"size(Width, Height)"
-----
Ok(
    Struct(
        Struct(
            Functor(
                :#0x2,
                2,
            ),
            [
                Variable(
                    :#0x0,
                ),
                Variable(
                    :#0x1,
                ),
            ],
        ),
    ),
)
